src/main/java/io/github/bobocodebreskul/config/PropertiesConfiguration.java

```diff
@@ -0,0 +1,105 @@
+package io.github.bobocodebreskul.config;
+
+import static java.util.Objects.isNull;
+
+import io.github.bobocodebreskul.context.exception.ConfigurationFileNotFoundException;
+import io.github.bobocodebreskul.context.exception.InvalidPropertyValueException;
+import io.github.bobocodebreskul.context.exception.LoadingPropertiesFailedException;
+import java.io.InputStream;
+import java.util.Properties;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * PropertiesConfiguration class stores properties from configuration file and provides an API to
+ * get property as string or integer value.
+ */
+@Slf4j
+public class PropertiesConfiguration {
+
+  private static String CONFIG_FILE = "application.properties";
+
+  private static Properties properties;
+
+  /**
+   * Method loads properties from configuration file and stores then in properties variable.
+   */
+  public static void loadProperties() {
+    Properties properties = new Properties();
+    try (InputStream input = PropertiesConfiguration.class.getClassLoader()
+        .getResourceAsStream(CONFIG_FILE)) {
+      if (input == null) {
+        log.info("Sorry, unable to find %s".formatted(CONFIG_FILE));
+        throw new ConfigurationFileNotFoundException(
+            "Sorry, unable to find %s".formatted(CONFIG_FILE));
+      }
```

Comment:
Is it expected to work on client side when this code is used as library? If yes then did you checked if it works as expected when this code is used as library?

Reply:
Yes, it works as a library